git_machete/github.py

```diff
@@ -125,14 +126,16 @@ def __fire_github_api_request(method: str, url: str, token: Optional[str], reque
     except HTTPError as err:
         if err.code == http.HTTPStatus.UNPROCESSABLE_ENTITY:
             raise UnprocessableEntityHTTPError(str(err.reason))
-        elif err.code in (http.HTTPStatus.UNAUTHORIZED, http.HTTPStatus.FORBIDDEN):
-            first_line = fmt(f'GitHub API returned {err.code} HTTP status with error message: `{err.reason}`\n')
+        elif err.code in (http.HTTPStatus.UNAUTHORIZED, http.HTTPStatus.FORBIDDEN, http.HTTPStatus.NOT_FOUND):
+            first_line = f'GitHub API returned {err.code} HTTP status with error message: `{err.reason}`\n'
             if token:
-                raise MacheteException(first_line + fmt(f'Make sure that the token provided in `gh auth status` or `~/.config/hub` or <b>{GITHUB_TOKEN_ENV_VAR}</b> is valid and allows for access to `{method.upper()}` https://{host}{url}`.'))
+                raise MacheteException(first_line + f'Make sure that the token provided in `gh auth status` or `~/.config/hub`'
+                                                    f' or <b>{GITHUB_TOKEN_ENV_VAR}</b> is valid and allows for access to `{method.upper()}` https://{host}{path}`.')
             else:
                 raise MacheteException(
-                    first_line + fmt(f'This repository might be private. Provide a GitHub API token with `repo` access via `gh` or `hub` or <b>{GITHUB_TOKEN_ENV_VAR}</b> env var.\n'
-                                     'Visit `https://github.com/settings/tokens` to generate a new one.'))
+                    first_line + f'You might not have the required permissions for this repository. '
```

Comment:
This might as well happen for a public repository... when you try pushing without a valid token